appinventor/blocklyeditor/src/warningHandler.js

```diff
@@ -386,10 +386,20 @@ Blockly.WarningHandler.prototype["checkIsNotInLoop"] = function(block) {
   }
 };
 
-Blockly_loopBlockTypes =
-  // add more later
-  ["controls_forEach", "controls_forRange", "controls_while"] ;
-
+// TODO: Maybe change this to a property of the block, instead of maintaining
+//   a list. Check how this interacts with extensions first.
+Blockly_loopBlockTypes = [
+    "controls_forEach",
+    "controls_for_each_dict",
+    "controls_forRange",
+    "controls_while"
+];
+
+// TODO: Maybe place this on an object. Options:
```

Comment:
I could see it being a property of a block, with the recursive call being made on the block's parent, if one exists. Potentially at that point we can memoize the result since most operations on the workspace won't change that (only moves that affect the ancestors of the current block).